source/guides/basics/mechanisms.md

```diff
@@ -1,49 +1,150 @@
 
 #### The AdjustBlock command
 
-The `adjustblock` command is a variant of normal `adjust` specifically to adjust `material` mechanisms on blocks.
+The `adjustblock` command is a variant of the normal `adjust` command. It's built specifically to adjust `material` mechanisms on blocks <span class="parens">([a list of which can be found here](https://one.denizenscript.com/denizen/mecs/materialtag))</span>.
+
+While the applicability of `adjustblock` is more niche, it can still be quite useful. For a quick usage example, you can use this script to force the plant block you are looking at to immediately change to its maximum age <span class="parens">(i.e. to cause newly-planted wheat to fully grow)</span>:
+
+```dscript_green
+grow_plant:
+    type: task
+    script:
+    - if <player.cursor_on.material.is_ageable>:
+        - adjustblock <player.cursor_on> age:<player.cursor_on.material.maximum_age>
+```
+
+As before, you can use this script by typing `/ex run grow_plant` in-game.
 
 #### How To Adjust Items
 
-**TODO: Explain adjust/adjustblock more, including how `inventory adjust` works, and including what a 'unique identifier' is**
+You can also adjust the properties of items in inventories. Doing so uses the `inventory` command with the `adjust` argument.
+
+You might think that the correct way to modify an item held by a player is by adjusting the item itself:
+
+```dscript_red
+magical_flint_bad:
+    type: world
+    events:
+        on player right clicks block with:flint:
+        - adjust <player.item_in_hand> material:diamond
+```
+
+However, this will only adjust the *generic concept* of the item in the player's hand. If you're not adjusting a specific item that exists in the world somehow, nothing is going to change in the world! Instead, you need to use the `adjust` argument of the `inventory` command to adjust the specific item that you want to change.
+
+The below example will *correctly* transform a player's flint into a diamond when they right click with it:
+
+```dscript_green
+magical_flint:
+    type: world
+    events:
+        on player right clicks block with:flint:
+        - inventory adjust slot:<player.held_item_slot> material:diamond
+```
+
+The `inventory adjust` command will specifically change the item at issue, because that's what it is designed to do! Because you identify the unique object of the item as it exists in the player's inventory, the `inventory adjust` command is therefore able to modify it as desired.
 
 #### Adjusting Definitions
 
-**TODO: Explain `adjust def:x`**
+The `adjust` command can also be used to modify an item stored in a definition. For example:
+
+```dscript_green
+defintion_adjust:
+    type: task
+    script:
+    - if <player.item_in_hand.material.name> != air:
+        - define player_item <player.item_in_hand>
+        - take <player.item_in_hand>
+        - adjust def:player_item "display_name:Adjusted by definition"
+        - give <[player_item]>
```

Comment:
This is explicitly an example of how NOT to script. It should not be used as a positive example.